E2E: rework the `Likes: Comments` spec.

worldomonation opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/wp-calypso/issues/74547.
Context: p1678991484142429/1678991483.901129-slack-CBTN58FTJ

## Proposed Changes

This PR reworks and introduces the use of API to set up and tear down the test environment.

Key changes:
- addition of new methods in the `RestAPIClient` class to support the deletion of posts, creation and deletion of comments, like and unlike of comments.
- refactor the spec to use API calls to set up a test post and comment, and to ensure consistent Like the state of the comment.

## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] Gutenberg E2E (desktop)
  - [x] Gutenberg E2E (mobile)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/secrets/encrypted.enc
- add more data for several users.

worldomonation committed:
packages/calypso-e2e/src/rest-api-client.ts
- add new methods to delete post, create and delete comments, and like/unlike comments.
- change interface names

worldomonation committed:
test/e2e/specs/plans/plans__upgrade.ts
- update interface name

test/e2e/specs/published-content/likes__comment.ts
- rework the spec to use RestAPIClient calls to set things up and clean up afterwards.

worldomonation committed:
test/e2e/specs/published-content/likes__comment.ts
- authenticate as the last task

worldomonation commented on packages/calypso-e2e/src/rest-api-client.ts:
```diff
@@ -723,6 +753,56 @@ export class RestAPIClient {
+	 * @param {number} siteID Target site ID.
+	 * @param {number} commentID Target comment ID.
+	 */
+	async commentAction(
```
In a slight break with convention, I've chosen to combine the action for comments, since the only thing that differed is the `endpoint` target as well as the post-processing to verify the action was performed as expected.

worldomonation commented on packages/calypso-e2e/src/types/rest-api-client.types.ts:
```diff
@@ -132,9 +132,11 @@ export interface DeleteInvitesResponse {
+	ID: number;
 	URL: string;
 	title: string;
+	status: 'trash' | 'publish' | 'private' | 'draft' | 'future' | 'deleted';
```
This is taken from https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/posts/delete/ and simplified.

worldomonation commented on test/e2e/specs/published-content/likes__comment.ts:
```diff
@@ -30,56 +28,86 @@ describe( DataHelper.createSuiteTitle( 'Likes (Comment) ' ), function () {
+
+		restAPIClient = new RestAPIClient( testAccount.credentials );
+
+		newPost = await restAPIClient.createPost(
+			testAccount.credentials.testSites?.primary.id as number,
+			{
+				title: DataHelper.getTimestamp() as string,
+				content: postContent,
+			}
+		);
+
+		newComment = await restAPIClient.createComment(
+			testAccount.credentials.testSites?.primary.id as number,
+			newPost.ID,
+			comment
+		);
```
The spec now uses API calls to set up the test post and comment.

I thought about using a locked and always-existing post and simply add a new comment to that, but I decided against that because there is no guarantee the "permanent" post will not be wiped out by an inadvertent headstart or theme change.

worldomonation replied:
After chatting with @fullofcaffeine in p1679099808423219/1678991483.901129-slack-CBTN58FTJ, an intermediate approach is to set up the test post programmatically (we already have many other specs that test the editor posting experience), but post comments and like/unlike using the UI.

worldomonation committed:
test/e2e/specs/published-content/likes__comment.ts
- reload the page for extra safety.

fullofcaffeine reviewed (approved):
LGTM. If this new approach eliminates the flakiness of this test, then I'm all for it!

One could argue that we should have a flow test for comments. Theare are core tests related to comments [here](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/comment.php), but it's more of an integration test starting from the controller layer, so maybe we should look into adding a test for comment flows somewhere? I wouldn't add it as a critical task, but it'd be nice. It's possible we might break stuff more often because of the Gutenberg upgrades (which core doesn't test that often).

karenroldan reviewed (commented):
I second the idea of programmatically creating a new post and a comment. I ran the Gutenberg tests (simple) and `likes__comment` passed. 👍🏻 

**issue:** when running the tests in Atomic, I am getting an error:

```
Error: unknown_blog: Unknown blog
at RestAPIClient.createPost
```

Perhaps we also need to update the secrets for the Atomic user.

worldomonation committed:
packages/calypso-e2e/src/secrets/encrypted.enc
- add Atomic user details

packages/calypso-e2e/src/lib/components/comments-component.ts
- scroll to the comment before acting on it
